### ISSUE TITLE
Bogus Warning with Optional.ofNullable

### ISSUE THREAD

**Issue by GoogleCodeExporter** [typetools/checker-framework#366]
```
What steps will reproduce the problem?
1.Run the Null Checker on the class below
package test;

import java.util.Optional;

class Test {
    static Optional<String> getPossiblyEmptyString() {
        String s = System.currentTimeMillis() % 2 == 0 ? null : "String";
        return Optional.ofNullable(s);
    }
}

What is the expected output? What do you see instead?
Expected: no error
Actual: test\Test.java:8: error: [return.type.incompatible] incompatible types 
in return.
        return Optional.ofNullable(s);
                                  ^
  found   : @Initialized @NonNull Optional<@Initialized @Nullable String>
  required: @Initialized @NonNull Optional<@Initialized @NonNull String>
1 error

What version of the product are you using? On what operating system?
1.8.6/Windows 8/Java 8

```

Original issue reported on code.google.com by `ClovisSe...@gmail.com` on 28 Sep 2014 at 9:24
